pyqg/tests/test_reference_solns.py

```diff
@@ -0,0 +1,244 @@
+import unittest
+import numpy as np
+import pyqg
+
+class ReferenceSolutionsTester(unittest.TestCase):
+
+    def test_two_layer(self):
+        """ Tests against some statistics of a reference two-layer solution """
+
+        year = 360*86400.
+        m = pyqg.QGModel(
+            nx=32,                      
+            L=1e6,                      
+            beta=1.5e-11,               
+            rek=5.787e-7,               
+            rd=30000.0,                 
+            delta=0.25,                 
+            U1=0.05,                    
+            U2=0.0,                     
+            filterfac=18.4,
+            dt=12800.,                   
+            tmax=3*year,          
+            tavestart=1*year,     
+            taveint=12800.,
+            useAB2=True,
+            diagnostics_list='all'     
+            )
+
+        m.set_q1q2(
+                (1e-6*np.cos(2*5*np.pi * m.x / m.L) +
+                 1e-7*np.cos(2*5*np.pi * m.y / m.W)),
+                np.zeros_like(m.x) )
+                    
+        m.run()
+
+        q1 = m.q[0] 
+        q1norm = (q1**2).sum()
+
+        assert m.t == 93312000.0
+   
+        
+        # machine + numpy.version fluctuations
+        #   appears to be less than 0.1%
+        rtol, atol = 0.001, 0.
+
+        np.testing.assert_allclose(q1norm, 9.561430503712755e-08, rtol=rtol, atol=atol,
+                    err_msg= ' Inconsistent with reference solution')
+
+        diagnostic_results = {
+                'APEgen': 2.5225558013107688e-07  / (m.nx**2),
+                'EKEdiss': 1.4806764171539711e-07 / (m.nx**2),           
+                }
```

Comment:
Wow, lots of stuff happened last night!

I see that ``APEgen`` and ``EKEdiss`` have now made it into the test. This is definitely an improvement.

There are now lots of commented out / skipped diagnostic tests below. Maybe in the comments we should just clarify what they are, where they came from, and whether we plan to fix them or drop them. For our own sanity in the future.